src/analysis/complex/conformal.lean

```diff
@@ -0,0 +1,136 @@
+begin
+  have minor₀ : ∀ (x : ℂ), x = x • 1 := λ x, by rw [smul_eq_mul, mul_one],
+  have minor₁ : ∥map 1∥ ≠ 0,
+  { contrapose! nonzero with w,
+    ext1,
+    rw [continuous_linear_map.zero_apply],
+    exact norm_eq_zero.mp w, },
+  refine ⟨∥map 1∥, minor₁, _⟩,
+  let li' := ∥map 1∥⁻¹ • map,
+  have key : ∀ (x : ℂ), ∥li' x∥ = ∥x∥,
+  { intros x,
+    simp only [li', continuous_linear_map.smul_apply],
+    nth_rewrite 0 [minor₀ x],
+    simp only [map.map_smul, norm_smul, normed_field.norm_inv, norm_norm],
+    field_simp [minor₁], },
+  let li : ℂ →ₗᵢ[ℝ] E := ⟨li', key⟩,
+  have minor₂ : (li : ℂ → E) = li' := rfl,
+  refine ⟨li, _⟩,
+  ext1,
+  simp only [minor₂, li', pi.smul_apply, continuous_linear_map.smul_apply, smul_smul],
+  field_simp [minor₁],
+end
+
+lemma is_conformal_map_conj : is_conformal_map conj_cle.to_continuous_linear_map :=
+⟨1, one_ne_zero, conj_lie.to_linear_isometry, by rw one_smul; refl⟩
+
+lemma is_conformal_map_complex_linear_conj
+  {map : ℂ →L[ℂ] E} (nonzero : map ≠ 0) :
+  is_conformal_map ((map.restrict_scalars ℝ).comp conj_cle.to_continuous_linear_map) :=
+is_conformal_map_conj.comp (is_conformal_map_complex_linear nonzero)
+
+end conformal_into_complex_normed
+
+section conformal_into_complex_plane
+
+open continuous_linear_map
+
+variables {f : ℂ → ℂ} {z : ℂ} {g : ℂ →L[ℝ] ℂ}
+
+lemma is_complex_or_conj_complex_linear (h : is_conformal_map g) :
+  (∃ (map : ℂ →L[ℂ] ℂ), map.restrict_scalars ℝ = g) ∨
+  (∃ (map : ℂ →L[ℂ] ℂ), map.restrict_scalars ℝ = g.comp conj_cle.to_continuous_linear_map) :=
+begin
+  rcases h with ⟨c, hc, li, hg⟩,
+  rcases linear_isometry_complex (li.to_linear_isometry_equiv rfl) with ⟨a, ha⟩,
+  let rot := c • (a : ℂ) • continuous_linear_map.id ℂ ℂ,
+  cases ha,
+  { refine or.intro_left _ ⟨rot, _⟩,
+    ext1,
+    simp only [coe_restrict_scalars', hg, ← li.coe_to_linear_isometry_equiv, ha,
+               pi.smul_apply, continuous_linear_map.smul_apply, rotation_apply,
+               continuous_linear_map.id_apply, smul_eq_mul], },
+  { refine or.intro_right _ ⟨rot, _⟩,
+    ext1,
+    rw [continuous_linear_map.coe_comp', hg, ← li.coe_to_linear_isometry_equiv, ha],
+    simp only [coe_restrict_scalars', function.comp_app, pi.smul_apply,
+               linear_isometry_equiv.coe_trans, conj_lie_apply,
+               rotation_apply, continuous_linear_equiv.coe_def_rev,
+               continuous_linear_equiv.coe_apply, conj_cle_apply],
+    simp only [continuous_linear_map.smul_apply, continuous_linear_map.id_apply,
+               smul_eq_mul, conj_conj], },
+end
+
+/-- A real continuous linear map on the complex plane is conformal if and only if the map or its
+    conjugate is complex linear, and the map is nonvanishing. -/
+lemma is_complex_or_conj_complex_linear_iff_is_conformal_map :
+  ((∃ (map : ℂ →L[ℂ] ℂ), map.restrict_scalars ℝ = g) ∨
+   (∃ (map : ℂ →L[ℂ] ℂ), map.restrict_scalars ℝ = g.comp conj_cle.to_continuous_linear_map))
+  ∧ g ≠ 0 ↔ is_conformal_map g :=
```

Comment:
Can you please swap the `iff` order?  It feels better to me with the high-level concept first.